### PR TITLE
Updated directory references, Clarified DB.Create.* documentation cle…

### PR DESCRIPTION
**Pull request by jameshensmancitrix**
…arer, Removed make_field type as to avoid it being distracting and added clarification to the ~get parameter needing to return a String

Signed-off-by: jameshensmancitrix <james.hensman@citrix.com>

### CODE REVIEW

File: devguide/howtos/add-field.md [hunk 1]
```diff
@@ -86,33 +86,20 @@ Example of a field in the pool class:
 
 See datamodel_types.ml for information about other parameters.
 
+## Changing Construcutors
```
Comment (psafont) [inline devguide/howtos/add-field.md:89]:
```suggestion
## Changing Constructors
```

File: devguide/howtos/add-field.md [hunk 2]
```diff
@@ -86,33 +86,20 @@ Example of a field in the pool class:
...
 Adding a field would change the constructors for the class – functions
 Db.*.create – and therefore, any references to these in the code need to be
 updated. In the example, the argument ~ha_enabled:false should be added to any
-call to Db.Pool.create.
+call to Db.Pool.create. 
+
+These calls will be in `ocaml/tests/common/test_common.ml` and `ocaml/xapi/xapi_[class].ml`.
 
-### ocaml/xapi/records.ml
+### ocaml/xapi-cli-server/records.ml
```
Comment (psafont) [inline devguide/howtos/add-field.md:98]: We probably don't want to use the filename as the title
```suggestion
### CLI Records
```

**Review by psafont — APPROVED** [review]

File: devguide/howtos/add-field.md [hunk 3]
```diff
@@ -56,8 +56,8 @@ new API fields used for ActiveDirectory integration were added:
 Adding the new field to some existing class
 -------------------------------------------
 
-## ocaml/idl/datamodel.ml
-Add a new "field" line to the class in the file `ocaml/idl/datamodel.ml`. The new field might require
+### ocaml/idl/datamodel.ml
+Add a new "field" line to the class in the file `ocaml/idl/datamodel_[class].ml`. The new field might require
```
Comment (robhoes) [inline devguide/howtos/add-field.md:60]: Actually, many of them are still in datamodel.ml

File: devguide/howtos/add-field.md [hunk 4]
```diff
@@ -86,33 +86,20 @@ Example of a field in the pool class:
...
 
+## Changing Constructors
+
 Adding a field would change the constructors for the class – functions
 Db.*.create – and therefore, any references to these in the code need to be
 updated. In the example, the argument ~ha_enabled:false should be added to any
-call to Db.Pool.create.
+call to Db.Pool.create. 
+
+These calls will be in `ocaml/tests/common/test_common.ml` and `ocaml/xapi/xapi_[class].ml`.
```
Comment (robhoes) [inline devguide/howtos/add-field.md:96]: "Many of these calls (...)". They may be in other files as well.

File: devguide/howtos/add-field.md [hunk 5]
```diff
@@ -2,19 +2,19 @@
...
-detect that an automatic database upgrade is necessary, or to find out that the
+detect that an automatic database upgrade is necessary or to find out that the
 new schema is incompatible with the existing database. If the schema version is
 not bumped, XAPI will start failing in unpredictable ways. Note that bumping
 the version is not necessary when adding functions, only when adding fields.
 
 The current version number is kept at the top of the file
-`ocaml/idl/datamodel.ml` in the variables `schema_major_vsn` and
+`ocaml/idl/datamodel_common.ml` in the variables `schema_major_vsn` and
 `schema_minor_vsn`, of which only the latter should be incremented (the major
```
Comment (edwintorok) [inline devguide/howtos/add-field.md:18]: Might be useful to mention that updating the API hash may be needed.

**Pull request merged**